src/authorization.ts

```diff
@@ -28,4 +28,11 @@ export interface Authorization {
      * Used to authorize Bundle transactions
      */
     isBundleRequestAuthorized(request: AuthorizationBundleRequest): Promise<boolean>;
+    /**
+     * Returns the resourceTypes for which the requester is allowed to perform the given operation.
+     */
+    resourceTypesAllowedForOperation(

```

Comment:
When would this be called by the router? ie we use `isAuthorized()` then we call `allowed`? Or are you thinking not every request?

Reply:
The call to isAuthorized() won't change. All requests go through it.

Right now only the router will only call `resourceTypesAllowedForOperation ` for search requests. The call can be done [here](https://github.com/awslabs/fhir-works-on-aws-routing/blob/mainline/src/router/routes/genericResourceRoute.ts#L99)

`isAuthorized()` can outright reject a request with 403 error. The `allowedResourceTypes` are used to narrow down search results, they don't cause `403` errors. i.e., 
searching for `/Patient?include=Patient:managingOrganization:Organization` with `allowedResourceTypes=['Patient', 'Practitioner']` will simply not include any Organization on the results.